### ISSUE TITLE
CI: GKE: Random failures in tests following K8sUpdates

### ISSUE THREAD

**Issue by pchaigno** [cilium/cilium#13898]
Since we unquarantined K8sUpdates, the GKE pipeline has been failing fairly consistently. The GKE test for the v1.9 backport PR that included the unquarantined K8sUpdates was also failing until we removed that commit.

It seems to always be tests happening after (although not always immediately after) K8sUpdates which fail (K8sUpdates succeeds). It seems likely that some resource shared by K8sUpdates and those failing tests is not properly cleaned after K8sUpdates runs.

Failing builds:
https://jenkins.cilium.io/job/Ginkgo-CI-Tests-Pipeline-GKE/2241/
    [45c73006_K8sKafkaPolicyTest_Kafka_Policy_Tests_KafkaPolicies.zip](https://github.com/cilium/cilium/files/5490569/45c73006_K8sKafkaPolicyTest_Kafka_Policy_Tests_KafkaPolicies.zip)
    [3764c836_K8sConformance_Portmap_Chaining_Check_one_node_connectivity-check_compliance_with_portmap_chaining.zip](https://github.com/cilium/cilium/files/5490570/3764c836_K8sConformance_Portmap_Chaining_Check_one_node_connectivity-check_compliance_with_portmap_chaining.zip)
    [af3f1185_K8sIstioTest_Istio_Bookinfo_Demo_Tests_bookinfo_inter-service_connectivity.zip](https://github.com/cilium/cilium/files/5490571/af3f1185_K8sIstioTest_Istio_Bookinfo_Demo_Tests_bookinfo_inter-service_connectivity.zip)
    [ccf928c9_K8sPolicyTest_GuestBook_Examples_checks_policy_example.zip](https://github.com/cilium/cilium/files/5490572/ccf928c9_K8sPolicyTest_GuestBook_Examples_checks_policy_example.zip)
    [e9d00d43_K8sConformance_Portmap_Chaining_Check_connectivity-check_compliance_with_portmap_chaining.zip](https://github.com/cilium/cilium/files/5490573/e9d00d43_K8sConformance_Portmap_Chaining_Check_connectivity-check_compliance_with_portmap_chaining.zip)
    [ee0305c2_K8sPolicyTest_Multi-node_policy_test_with_L7_policy_using_connectivity-check_to_check_datapath.zip](https://github.com/cilium/cilium/files/5490574/ee0305c2_K8sPolicyTest_Multi-node_policy_test_with_L7_policy_using_connectivity-check_to_check_datapath.zip)
https://jenkins.cilium.io/job/Ginkgo-CI-Tests-Pipeline-GKE/2234/
https://jenkins.cilium.io/job/Ginkgo-CI-Tests-Pipeline-GKE/2231/
https://jenkins.cilium.io/job/Ginkgo-CI-Tests-Pipeline-GKE/2230/
https://jenkins.cilium.io/job/Ginkgo-CI-Tests-Pipeline-GKE/2229/
https://jenkins.cilium.io/job/Ginkgo-CI-Tests-Pipeline-GKE/2228/
https://jenkins.cilium.io/job/Ginkgo-CI-Tests-Pipeline-GKE/2227/
https://jenkins.cilium.io/job/Ginkgo-CI-Tests-Pipeline-GKE/2226/
https://jenkins.cilium.io/job/Ginkgo-CI-Tests-Pipeline-GKE/2222/
No artifacts:
https://jenkins.cilium.io/job/Ginkgo-CI-Tests-Pipeline-GKE/2235/
https://jenkins.cilium.io/job/Ginkgo-CI-Tests-Pipeline-GKE/2223/

Set of tests that fail in the above builds:
```
3 Suite-k8s-1.16.K8sChaosTest Connectivity demo application Endpoint can still connect while Cilium is not running
2 Suite-k8s-1.16.K8sConformance Portmap Chaining Check connectivity-check compliance with portmap chaining
2 Suite-k8s-1.16.K8sConformance Portmap Chaining Check one node connectivity-check compliance with portmap chaining
1 Suite-k8s-1.16.K8sDatapathConfig DirectRouting Check connectivity with direct routing
1 Suite-k8s-1.16.K8sDatapathConfig DirectRouting Check connectivity with direct routing and endpointRoutes
1 Suite-k8s-1.16.K8sDatapathConfig MonitorAggregation Checks that monitor aggregation flags send notifications
1 Suite-k8s-1.16.K8sDatapathConfig MonitorAggregation Checks that monitor aggregation restricts notifications
1 Suite-k8s-1.16.K8sDemosTest Tests Star Wars Demo
3 Suite-k8s-1.16.K8sIstioTest Istio Bookinfo Demo Tests bookinfo inter-service connectivity
4 Suite-k8s-1.16.K8sKafkaPolicyTest Kafka Policy Tests KafkaPolicies
1 Suite-k8s-1.16.K8sPolicyTest GuestBook Examples checks policy example
3 Suite-k8s-1.16.K8sPolicyTest Multi-node policy test with L7 policy using connectivity-check to check datapath
1 Suite-k8s-1.16.K8sServicesTest Bookinfo Demo Tests bookinfo demo
```
At a glance, it's not clear what resources those tests may have in common.

**Comment by brb** [comment]
@gandro Have you fixed this?

**Comment by gandro** [comment]
> @gandro Have you fixed this?

I think so. Unfortunately, we have not yet been able to quarantine it yet. 
The fix of the underlying issue described here landed in #14187 - see https://github.com/cilium/cilium/issues/13833#issuecomment-733809602 for the in-depth description.

However, right after the fix landed, PR #13878 broke the preflight check on GKE, thus also breaking the upgrade test. That second one needed two follow-up fixes, one of which is still outstanding: #14295 + #14393
